lucene/core/src/java/org/apache/lucene/index/NoMergePolicy.java

```
@@ -39,6 +40,11 @@ public MergeSpecification findMerges(
     return null;
   }
 
+  @Override
+  public MergeSpecification findMerges(List<CodecReader> readers) throws IOException {
+    return null;
```

Hmm should `null` even be allowed by this new API?  Won't this throw `NPE` if someone tries to `addIndexes`?

---

In my current impl, addIndexes simply completes without adding anything if the API returns null. I now realize, this is misleading for the user. Callers of addIndexes may not know about the configured merge policy. I should probably throw an exception if I get a null here?
On similar lines, what should we do if the API call had some CodecReaders, but findMerges returns an empty spec.? Should we error out? (Will add a log for this either way.)

Re: letting the API return null, I see that other variants of `findMerges` in this policy, all return `null`, and all callers of findMerges seem to be doing the null checks. Personally, I prefer disallowing the null return values, but I wanted to keep this in sync with the other variants.

---

There are consumers of `NoMergePolicy` that expect `addIndexes(CodecReaders...)` to work, like `IndexRearranger`. My understanding is that NoMergePolicy is meant to turn off any merges on existing segments. Since this API uses merges to add new readers to the index, I've changed it here to return the default base class impl.

It's a little off because the name is NoMergePolicy, but I think(hope) it aligns with the intent behind that class. I've added some comments to call this out. Let me know if we should instead create a different MergePolicy that only allows fineMerges(readers).. (it would be exactly like NoMergePolicy except for one api).